page_analyzer/utils/db.py

```diff
@@ -0,0 +1,90 @@
+import os
+from typing import Any, Dict, List
+
+import psycopg2
+from dotenv import load_dotenv
+
+load_dotenv()
+
+
+def connect_db() -> Any:
+    conn = psycopg2.connect(os.getenv("DATABASE_URL"))
+    return conn
+
+
+def get_url_context(url_id: int) -> Dict[str, str]:
+    conn = connect_db()
+    cur = conn.cursor()
+    cur.execute(
+        "SELECT id FROM urls WHERE id=%(int)s;",
+        {
+            "int": url_id,
+        },
+    )
+    target_id = cur.fetchone()
+    if target_id:
+        cur.execute(
+            "SELECT name, created_at FROM urls WHERE id=%(int)s;",
+            {
+                "int": url_id,
+            },
+        )
+        page_url, created_at = cur.fetchone()
+        cur.close()
+        conn.close()
+        context = {
+            "id": str(url_id),
+            "name": str(page_url),
+            "created_at": str(created_at),
+        }
+        return context
+    cur.close()
+    conn.close()
+    context = {}
+    return context
+
+
+def check_url_existence(url_name: str) -> List[bool | int]:
+    conn = connect_db()
+    cur = conn.cursor()
+    cur.execute(
+        "SELECT id name FROM urls WHERE name=%(str)s;",
+        {
+            "str": url_name,
+        },
+    )
+    id_tuple = cur.fetchone()
+    if id_tuple:
+        url_id = id_tuple[0]
+        cur.close()
+        conn.close()
+        return [True, int(url_id)]
+    cur.close()
+    conn.close()
+    return [False]
```

Comment:
same here, can be simplified
wdyt about returning URL ID or None if it doesn't exist?